[CI] NoSecurityManagerTests testPrepopulateSecurityCaller failing

**Build Scans:**
- [elasticsearch-periodic #6321 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/ugdu5yhetykuu)
- [elasticsearch-periodic #6314 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/onqxomtzadmuu)
- [elasticsearch-periodic #6307 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/o5deishzcppbq)
- [elasticsearch-periodic #6300 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/lrhq2cllc6xsq)
- [elasticsearch-periodic #6293 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/h75prbvbxfdz6)
- [elasticsearch-periodic #6279 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/uzxuupok7f2su)
- [elasticsearch-periodic #6272 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/5t3wp3ne2rzom)
- [elasticsearch-periodic #6251 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/2lcpjsgt7vc56)
- [elasticsearch-periodic #6237 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/jjkukr4usxm3y)
- [elasticsearch-periodic #6223 / openjdk24_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/qqwas4uxlb2b4)

**Reproduction Line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.bootstrap.NoSecurityManagerTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testPrepopulateSecurityCaller),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: is <true>
     but: was <false>
```

**Issue Reasons:**
- [main] 11 consecutive failures in test testPrepopulateSecurityCaller
- [main] 11 consecutive failures in step openjdk24_checkpart1_java-matrix
- [main] 11 consecutive failures in pipeline elasticsearch-periodic
- [main] 11 failures in test testPrepopulateSecurityCaller (100.0% fail rate in 11 executions)
- [main] 11 failures in step openjdk24_checkpart1_java-matrix (100.0% fail rate in 11 executions)
- [main] 11 failures in pipeline elasticsearch-periodic (100.0% fail rate in 11 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/test/java/org/elasticsearch/bootstrap/NoSecurityManagerTests.java
+++ b/server/src/test/java/org/elasticsearch/bootstrap/NoSecurityManagerTests.java
@@ -12,6 +12,7 @@ package org.elasticsearch.bootstrap;
 import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
 
 import org.apache.lucene.tests.util.LuceneTestCase;
+import org.elasticsearch.jdk.RuntimeVersionFeature;
 import org.elasticsearch.test.GraalVMThreadsFilter;
 
 import static org.hamcrest.Matchers.is;
@@ -20,6 +21,7 @@ import static org.hamcrest.Matchers.is;
 public class NoSecurityManagerTests extends LuceneTestCase {
 
     public void testPrepopulateSecurityCaller() {
+        assumeTrue("security manager must be available", RuntimeVersionFeature.isSecurityManagerAvailable());
         assumeTrue("Unexpected security manager:" + System.getSecurityManager(), System.getSecurityManager() == null);
         boolean isAtLeastJava17 = Runtime.version().feature() >= 17;
         boolean isPrepopulated = Security.prepopulateSecurityCaller();

